### ISSUE TITLE
[BUG] Volume can't turn into healthy when upgrading from v1.0.3 to v1.1.0-rc2 (master build)

### ISSUE THREAD

**Issue by bk201** [harvester/harvester#2920]
**Describe the bug**

We saw a volume that can't become healthy during an upgrade. One of the replicas keeps being deleted and rebuilt.
<img width="1897" alt="Screen Shot 2022-10-12 at 10 41 39 AM" src="https://user-images.githubusercontent.com/1691518/195291052-fa19adcb-1729-4c2d-b13f-51e0eef33f06.png">

**To Reproduce**
1. Upgrade Harvester from v1.0.3 to master build (with LH v1.3.2-rc2).
2. The first node is upgraded and rebooted.
3. The pre-drain job in the next node gets stuck due to volumes that can't turn into healthy.

**Expected behavior**

Volumes should be healthy in this situation.

**Support bundle**

**Environment**
 - Harvester ISO version: 
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**
From @mantissahz, this could be caused by https://github.com/longhorn/longhorn/issues/4705

**Comment by bk201** [comment]
https://github.com/longhorn/longhorn/issues/4705 is closed, and the issue is supposed to be included in the v1.3.2 formal release.

**Comment by Vicente-Cheng** [comment]
related error message:

```
2022-10-13T10:15:49.301431337Z stderr F time="2022-10-13T10:15:49Z" level=error msg="Failed rebuilding of replica 10.52.0.155:10060" controller=longhorn-engine engine=pvc-b7e36bb2-e14b-492a-9b2c-31ab140b689c-e-d7ea1277 error="failed to add replica address='tcp://10.52.0.155:10060' to controller 'pvc-b7e36bb2-e14b-492a-9b2c31ab140b689c': failed to execute: /var/lib/longhorn/engine-binaries/longhornio-longhorn-engine-v1.3.2-rc2/longhorn [--url 10.52.1.22:10001 add tcp://10.52.0.155:10060], output , stderr, time=\"2022-10-13T10:15:49Z\" level=fatal msg=\"Error running add replica command: size is required\"\n, error exit status 1" node=harvester-node-1 volume=pvc-b7e36bb2-e14b-492a-9b2c-31ab140b689c
2022-10-13T10:15:49.301555781Z stderr F time="2022-10-13T10:15:49Z" level=info msg="Event(v1.ObjectReference{Kind:\"Engine\", Namespace:\"longhorn-system\", Name:\"pvc-b7e36bb2-e14b-492a-9b2c-31ab140b689c-e-d7ea1277\", UID:\"aa02d770-2d6a-4e79-846f-12e870eedbe4\", APIVersion:\"longhorn.io/v1beta2\", ResourceVersion:\"139456\", FieldPath:\"\"}): type: 'Warning' reason: 'FailedRebuilding' Failed rebuilding replica with Address 10.52.0.155:10060: failed to add replica address='tcp://10.52.0.155:10060' to controller 'pvc-b7e36bb2-e14b-492a-9b2c-31ab140b689c': failed to execute: /var/lib/longhorn/engine-binaries/longhornio-longhorn-engine-v1.3.2-rc2/longhorn [--url 10.52.1.22:10001 add tcp://10.52.0.155:10060], output , stderr, time=\"2022-10-13T10:15:49Z\" level=fatal msg=\"Error running add replica command: size is required\"\n, error exit status 1"
2022-10-13T10:15:49.310176022Z stderr F time="2022-10-13T10:15:49Z" level=info msg="Removed failed rebuilding replica 10.52.0.155:10060" controller=longhorn-engine engine=pvc-b7e36bb2-e14b-492a-9b2c-31ab140b689c-e-d7ea1277 node=harvester-node-1 volume=pvc-b7e36bb2-e14b-492a-9b2c-31ab140b689c
```

According to above error message, we could see the volume `pvc-b7e36bb2-e14b-492a-9b2c-31ab140b689c` rebuild fail.
As the following figure.
<img width="1425" alt="截圖 2022-10-13 下午6 19 35" src="https://user-images.githubusercontent.com/1615081/195572718-eed9a8ac-e164-4901-b657-44467c75b79b.png">

The volume could not rebuild successfully so the upgrade would be stuck (now, we need to wait for the volume healthy to continue).

**Comment by TachunLin** [comment]
Verified **fixed** on `v1.1.0-rc3`. Close this issue.

### Result
* Can finish the pre-drain of each node and successfully upgrade to v1.1.0-rc3
![image](https://user-images.githubusercontent.com/29251855/196434398-a61b5111-7723-4fa6-ac57-2a68ffef73ee.png)

* Volume status after upgrade in Longhorn UI did not have `degraded` status
![image](https://user-images.githubusercontent.com/29251855/196433439-90b9fee5-0c20-4216-b91d-a866a27f9c62.png)

* The pvc volume status after upgrade
![image](https://user-images.githubusercontent.com/29251855/196433662-b3add848-9f82-4277-83a8-e13df352b09d.png)

### Test Information
* Test Environment: 4 nodes harvester on provo bare machine
* Harvester version: v1.1.0-rc3

### Verify Steps
1. Prepare a 4 nodes v1.0.3 Harvester cluster 
2. Install several images
3. Create three VMs
4. Enable Network 
5. Create vlan1 network
6. Shutdown all VMs
7. Upgrade to v1.1.0-rc3
